lms/djangoapps/instructor/views/api.py

```diff
@@ -1228,6 +1228,29 @@ def get_enrollment_report(request, course_id):
         })
 
 
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
+@require_finance_admin
+def get_exec_summary_report(request, course_id):
+    """
+    get the executive summary report for the particular course.
+    """
+    course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    try:
+        instructor_task.api.submit_executive_summary_report(request, course_key)
+        status_response = _("Your Executive Summary report is being generated! "
+                            "You can view the status of the generation task in"
+                            " the 'Pending Instructor Tasks' section.")
+    except AlreadyRunningError:
+        status_response = _("An Executive Summary report generation task is already in progress. "
+                            "Check the 'Pending Instructor Tasks' table for the status of the task. "
+                            "When completed, the report will be available for download in the table below.")
```

Comment:
An executive summary report is currently in progress.
To view the status of the report, see the "Pending Instructor Tasks" section. You will be able to download the report when it is complete.